core/embed/lib/sizedefs.h

```diff
@@ -24,4 +43,54 @@
 #define SIZE_256M (256 * 1024 * 1024)
 #define SIZE_512M (512 * 1024 * 1024)
 
-#endif
+// Checks if a value 'x' is a power of two.
+#define IS_POWER_OF_TWO(x) (((x) != 0) && (((x) & ((x)-1)) == 0))
+
+// Ensures at compile-time that 'x' is a power of two.
+#define ENSURE_POWER_OF_TWO(x) \
+  _Static_assert(IS_POWER_OF_TWO(x), "Value must be a power of two")
+
+// Checks if 'addr' is properly aligned to 'align', which must be a
+// power of two.
+#define IS_ALIGNED(addr, align)    \
+  ({                               \
+    ENSURE_POWER_OF_TWO(align);    \
+    (((addr) & ((align)-1)) == 0); \
+  })
+
+// Ensures that that 'addr' is properly aligned to 'align', which
+// must be a power of two.
+#define ENSURE_ALIGNMENT(addr, align) \
+  _Static_assert(IS_ALIGNED(addr, align), "Address must be aligned")
+
+// Aligns 'addr' upwards to the next boundary of 'align', which must
+// be a power of two.
+#define ALIGN_UP(addr, align)              \
+  ({                                       \
+    ENSURE_POWER_OF_TWO(align);            \
+    (((addr) + (align)-1) & ~((align)-1)); \
+  })
+
+// Aligns 'addr' upwards to the next boundary of 'align', which must
+// be a power of two.
+//
+// This version is for use in constant expressions. Use only if `ALIGN_DOWN`
```

Comment:
Yes, the name might be a bit misleading in this case. 

 `ALIGN_UP`/`ALIGN_DOWN` should be used wherever possible. However, it's somewhat restrictive since it uses a "statement expression", which cannot be used, for example, in declarations. `ALIGN_UP_CONST`/`ALIGN_DOWN_CONST` was introduced to overcome this limitation, allowing usage in contexts where constant expressions are expected, if the provided is a constant expression. Therefore the '_CONST' suffix in the name. 

I hadn’t realized that it could also be used, as you suggested, with non-constant `align` argument. But I would say that such a cases are very rare.

I’ve at least adjusted the comment, see f31168508f7e2927b35368011c82bbe83065d97e